go/mcap/indexed_message_iterator.go

```diff
@@ -237,13 +238,54 @@ func (it *indexedMessageIterator) loadChunk(chunkIndex *ChunkIndex) error {
 	return nil
 }
 
+func readRecord(r io.Reader) (TokenType, []byte, error) {
```

Comment:
surprising there wasn't already a function for this somewhere

Reply:
agreed